lib/cluster.js

```diff
@@ -494,15 +496,35 @@ function masterInit() {
     if (!handle.data) handle.data = message.data;
 
     // Set custom server data
-    handle.add(worker, function(errno, reply, handle) {
+    handle.add(worker, function(errno, reply, handle_) {
+      var data;
+      if (port === 0) {
+        delete handles[key];
+        key = [message.address,
+               reply && reply.sockname && reply.sockname.port || message.port,
+               message.addressType,
+               message.fd,
+               message.index].join(':');
+        if (!errno)
+          handles[key] = handle;
+        data = handle.data;
+        handle.key = key;
```

Comment:
It shouldn't really matter. I just did it there so that it's set no matter if there should happen to be another reference to the handle now or in the future.